GAD 6 - Verbessertes Sortieren/tests/simplesort/PivotFinderTester.java

```diff
@@ -241,25 +241,37 @@ public void midPivotTest() {
      * @author Aamin
+     * @version 1.0.3
      */
     @Test
     public void randomPivot() {
         int[] array;
+        int answerGot;
 
         array = new int[] {123, 4123, 5290, 109, -123, 123};
         PivotFinder pivotFinder = PivotFinder.getRandomPivot(69420);
 
         assertEquals(4, pivotFinder.findPivot(array, 0, 5));
-        assertEquals(3, pivotFinder.findPivot(array, 0, 5));
-        assertEquals(2, pivotFinder.findPivot(array, 0, 5));
-        assertEquals(4, pivotFinder.findPivot(array, 0, 5));
-        assertEquals(5, pivotFinder.findPivot(array, 0, 5));
-        assertEquals(1, pivotFinder.findPivot(array, 1, 5));
-        assertEquals(4, pivotFinder.findPivot(array, 2, 5));
-        assertEquals(5, pivotFinder.findPivot(array, 3, 5));
-        assertEquals(3, pivotFinder.findPivot(array, 2, 4));
-        assertEquals(3, pivotFinder.findPivot(array, 2, 3));
-        assertEquals(3, pivotFinder.findPivot(array, 3, 3));
+        answerGot = pivotFinder.findPivot(array, 0, 5);
+        assertEquals(3, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 0, 5);
+        assertEquals(2, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 0, 5);
+        assertEquals(4, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 0, 5);
+        assertEquals(5, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 1, 5);
+        assertEquals(1, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 2, 5);
+        assertEquals(4, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 3, 5);
+        assertEquals(5, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 2, 4);
+        assertEquals(3, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 2, 3);
+        assertEquals(3, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
+        answerGot = pivotFinder.findPivot(array, 3, 3);
+        assertEquals(3, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
```

Comment:
I think I see a lot of copy & paste here, don't I? :)
Every string is equal, especially the `expected` part:

> `Expected <3> got <" + answerGot + ">"`

Please fix those errors introduced by e6b1457d252f675393c81bb56939123321342c41.

Reply:
I thought about it, but my argument was, that if I try removing the redundant part, it'd make the code a bit obfuscated, which means that other students need some time to understand my code, whereas if the code is at a level of a 5y/o, direct plain copy paste, although it doesn't look "professional", it is a lot easier to understand what is happening.

Reply:
Even though it's not very beautiful, I have no problem with the same string. Even though it could be improved easily.
My point is the `<3>`. In every sentence, there is the static `<3>`. Because sometimes different values were expected. Example:

```java
answerGot = pivotFinder.findPivot(array, 3, 5);
assertEquals(5, answerGot, "Are you sure you aren't creating new object of random every time? Expected <3> got <" + answerGot + ">");
```

You actually expect the result to be **5** and not **3**, but every string contains _expected 3_!

Reply:
Oh shoot! I'm blind! Hahaha, my bad, thanks a lot for reviewing it. Gonna fix it asap!